### PR TITLE
Move Dapps files from @parity/shared ; display icons of builtin dapps from filesystem

### PR DESCRIPTION
**Pull request by axelchalon**
- Merge dappsViews onto dappsBuiltin
- Move dapps files from `@parity/shared` to the shell
- Fetch icons of builtin dapps from filesystem using manifest.json

### CODE REVIEW

**Comment by axelchalon** [comment]
Build is failing for some strange reason. I don't have the rights to relaunch the build.

Uses https://github.com/parity-js/ui/pull/9. Waiting for it to be merged so that I can update the `@parity/ui` dependency.

**Comment by amaurymartiny** [comment]
I have one question: if I use DappIcon inside a dapp (dapp-visible or dapp-methods use it), will this still work?

For the window.require, maybe try if isElectron(), then window.require or else require.

**Comment by axelchalon** [comment]
:+1: Thanks for the tip, tests work fine now!

Good question! It seems to be broken inside the dapps at the moment; I'll look into this.

**Comment by axelchalon** [comment]
dapp-visible and dapp-methods should work fine now; a bit tricky to test locally because dev dapps are served through http and don't have access to the fs -- but as far as I tested they work fine. We'll just have to update their `@parity/ui` dependency.

**Comment by axelchalon** [comment]
dapp-visible and dapp-methods work fine here! PR ready for review/merge

File: src/util/dapps.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+export function fetchBuiltinApps (api) {
+  const basePath = isElectron() ? window.require('electron').remote.getGlobal('dirName') : path.join(__dirname, '..');
+  const initialApps = builtinApps.filter(app => app.id);
+
+  // Replace all backslashes by front-slashes (happens in Windows)
+  // Note: `dirName` contains backslashes in Windows. One would assume that
+  // path.join in Windows would handle everything for us, but after some time
+  // I realized that even in Windows path.join here bahaves like POSIX (maybe
+  // it's electron, maybe browser env?). Switching to '/'. -Amaury 12.03.2018
+  const posixDirName = basePath.replace(/\\/g, '/');
```
Comment (amaurymartiny) [inline src/util/dapps.js]: This piece of code is now in 2 places. We could add another file inside `util/` to factorize that.

**Pull request merged**